File: AD-HOC/1515/main.dart
import 'dart:io';

String firstCivilization(int n) {
  Map<int, String> messages = {};
  for (int times = 0; times < n; times++) {
    var [civilization, received, traveled] = stdin.readLineSync()!.split(' ');
    messages[int.parse(received) - int.parse(traveled)] = civilization;
  }
  List<int> years = messages.keys.toList()..sort();
  return messages[years.first]!;
}

void main() {
  int n;
  while ((n = int.parse(stdin.readLineSync()!)) > 0) {
    print(firstCivilization(n));
  }
}

File: AD-HOC/1515/main.rb
# frozen_string_literal:true

def first_civilization(n)
  messages = {}
  n.times do
    civilization, received, traveled = gets.split
    messages[received.to_i - traveled.to_i] = civilization
  end
  smallest_year = messages.keys.min
  messages[smallest_year]
end

while (n = gets.to_i).positive?
  puts first_civilization(n)
end
